src/Workspaces/Core/Portable/Workspace/Solution/SourceGeneratedDocumentIdentity.cs

```diff
@@ -16,8 +17,12 @@ namespace Microsoft.CodeAnalysis
     /// A small struct that holds the values that define the identity of a source generated document, and don't change
     /// as new generations happen. This is mostly for convenience as we are reguarly working with this combination of values.
     /// </summary>
-    internal readonly record struct SourceGeneratedDocumentIdentity
-        (DocumentId DocumentId, string HintName, SourceGeneratorIdentity Generator, string FilePath)
+    [DataContract]
+    internal readonly record struct SourceGeneratedDocumentIdentity(
+        [property: DataMember(Order = 0)] DocumentId DocumentId,
+        [property: DataMember(Order = 1)] string HintName,
+        [property: DataMember(Order = 2)] SourceGeneratorIdentity Generator,
+        [property: DataMember(Order = 3)] string FilePath)
```

Comment:
this is just preference.  i prefer primary constructor style records to property-style ones.